[EDD] 1주차 wootube 기능 구현

kimjungmin-developer opened the pull request:
안녕하세요! 저희는 wootube를 만들고 있는 EDD(Everyone-Driven-Development)입니다! 우선 1주차 기능구현 pr보냅니다! 잘 부탁드립니다!

wootube 1주차 기능구현 목록:

- 배포환경 구축
- 동영상 저장 API 구현
- 메인페이지 구현
- 최신 동영상 리스트 조회 API 구현
- 메인페이지 연결 API 구현
- 동영상 등록 페이지 구현
- 동영상 상세 페이지 구현
- 동영상 상세 API 구현

kimjungmin-developer committed:
initial commit

kangmin46 committed:
[feat] add main page markup (#11)

* [docs] add index.html, static resourse

* [feat] add videocardtemplate.js

* [feat]  add const.js

* [feat] add videoCardTemplate method

* [feat] add mainpage event

* [feat] add popular video card section

*  최신순으로 video들 가져오는 메서드 구현

* create controllerTest

* [feat] add backend code

* [refactor] detach fetch api

* [chore] delete backend directory

* [chore] delete .idea directory

* [refactor] change directory structure

* [refactor] change directory structure

* [refactor] Change path js file

heebong committed:
[feat] add save video api (#10)

* [docs] add index.html, static resourse

* [feat] add videocardtemplate.js

* [chore] junit 5 setting

* [feat] add vo

- Contents, Title, YoutubeId

* [feat] add video entity

* [chore] add database

* [refactor] move column annotation

* [feat]  add const.js

* [feat] add videoCardTemplate method

* [feat] add mainpage event

* [feat] add popular video card section

* [chore] add mockito

* [feat] add POST /videos method

*  최신순으로 video들 가져오는 메서드 구현

* create controllerTest

* [feat] add backend code

* [feat] add video error object

* [refactor] change package structure

* [refactor] remove setter

* [refactor] add RequestMapping

* [refactor] use ResponseEntity

* [refactor] add Contents exception

* [refactor] change exception static value name

* [refactor] add repository annotation

* [refactor] add autowired annotation

* [chore] add trace, debug setting

* [style] formatting

* [feat] add main page markup (#11)

* [docs] add index.html, static resourse

* [feat] add videocardtemplate.js

* [feat]  add const.js

* [feat] add videoCardTemplate method

* [feat] add mainpage event

* [feat] add popular video card section

*  최신순으로 video들 가져오는 메서드 구현

* create controllerTest

* [feat] add backend code

* [refactor] detach fetch api

* [chore] delete backend directory

* [chore] delete .idea directory

* [refactor] change directory structure

* [refactor] change directory structure

* [refactor] Change path js file

* [feat] add main app

Laterality committed:
[chore] set deploy environment(#14)

* Add scripts

- Added initializing script.
- Added docker  image pushing script.
- Added nginx configuration for nginx container.

* Add travis configuration

* refactor: update initializing script remove unnecessary code

* chore: add execution permission to gradlew

* chore: update travis configuration

- Changed branch to trigger build.

* chore: update travis configuration

- Changed branch pattern to trigger build.

* chore: update travis configuration

- Changed build triggering branch name pattern.

* chore: update with travis ci integration

* Added dockerfile for frontend/backend image.
* Added docker-compose file with production/development.
* Added script for each stage(build jar, docker build, docker push).
* Added script for initializing server.
* Added nginx configuration.

* chore: change initialize script's download url temporarily

* chore: update initialzing script typo

* chore: update docker compose file to run nginx container after running app container

* chore: reduce docker compose file

- Updated initializing script.

* chore: update initializing script

* chore: update init script to clone with specific branch

* chore: update dockerfile

* chore: update dockerfile

* chore: update docker compose file typo

* chore: remove unnecessary files

* chore: update docker-compose frontend root directory

* chore: remove unnecessary code

* chore: move gitignore

Laterality committed:
[chore] add setting file (#15)

* Add scripts

- Added initializing script.
- Added docker  image pushing script.
- Added nginx configuration for nginx container.

* Add travis configuration

* refactor: update initializing script remove unnecessary code

* chore: add execution permission to gradlew

* chore: update travis configuration

- Changed branch to trigger build.

* chore: update travis configuration

- Changed branch pattern to trigger build.

* chore: update travis configuration

- Changed build triggering branch name pattern.

* chore: update with travis ci integration

* Added dockerfile for frontend/backend image.
* Added docker-compose file with production/development.
* Added script for each stage(build jar, docker build, docker push).
* Added script for initializing server.
* Added nginx configuration.

* chore: change initialize script's download url temporarily

* chore: update initialzing script typo

* chore: update docker compose file to run nginx container after running app container

* chore: reduce docker compose file

- Updated initializing script.

* chore: update initializing script

* chore: update init script to clone with specific branch

* chore: update dockerfile

* chore: update dockerfile

* chore: update docker compose file typo

* chore: remove unnecessary files

* chore: update docker-compose frontend root directory

* chore: remove unnecessary code

* chore: move gitignore

* chore: update gitignore and missed gradle related files

Conatuseus committed:
[feat] add retrieve videos API (#16)

*  최신순으로 video들 가져오는 메서드 구현

* create controllerTest

* add test code

* Fix date to constant

* fix import

* fix import

* fix contents exceptions

* [refactor] fix structure of packages

* [refactor] fix database to mySql

* [refactor] fix value to constant

* [refactor] fix value to constant

Conatuseus committed:
[fix] Controller url 수정 (#17)

*  최신순으로 video들 가져오는 메서드 구현

* create controllerTest

* add test code

* Fix date to constant

* fix import

* fix import

* fix contents exceptions

* [refactor] fix structure of packages

* [refactor] fix database to mySql

* [refactor] fix value to constant

* [refactor] fix value to constant

* [fix] change controller mapping url

kangmin46 committed:
[refactor] delete window.onloadmethod, localhost url (#18)

* [docs] add index.html, static resourse

* [feat] add videocardtemplate.js

* [feat]  add const.js

* [feat] add videoCardTemplate method

* [feat] add mainpage event

* [feat] add popular video card section

*  최신순으로 video들 가져오는 메서드 구현

* create controllerTest

* [feat] add backend code

* [refactor] detach fetch api

* [chore] delete backend directory

* [chore] delete .idea directory

* [refactor] change directory structure

* [refactor] change directory structure

* [refactor] Change path js file

* [fix] delete localhost url

* [refactor] delete window.onload method

Laterality committed:
Revert "[refactor] delete window.onloadmethod, localhost url (#18)" (#19)

This reverts commit d2ea40d1c4fa5b9681115c30f5dc92f75ede0518.

kangmin46 committed:
[refactor] delete localhost url, window onload (#20)

Laterality committed:
[chore] fix deploy and restarting problem (#21)

* Add scripts

- Added initializing script.
- Added docker  image pushing script.
- Added nginx configuration for nginx container.

* Add travis configuration

* refactor: update initializing script remove unnecessary code

* chore: add execution permission to gradlew

* chore: update travis configuration

- Changed branch to trigger build.

* chore: update travis configuration

- Changed branch pattern to trigger build.

* chore: update travis configuration

- Changed build triggering branch name pattern.

* chore: update with travis ci integration

* Added dockerfile for frontend/backend image.
* Added docker-compose file with production/development.
* Added script for each stage(build jar, docker build, docker push).
* Added script for initializing server.
* Added nginx configuration.

* chore: change initialize script's download url temporarily

* chore: update initialzing script typo

* chore: update docker compose file to run nginx container after running app container

* chore: reduce docker compose file

- Updated initializing script.

* chore: update initializing script

* chore: update init script to clone with specific branch

* chore: update dockerfile

* chore: update dockerfile

* chore: update docker compose file typo

* chore: remove unnecessary files

* chore: update docker-compose frontend root directory

* chore: remove unnecessary code

* [chore]: add gitignore

* chore: move gitignore

* chore: update gitignore and missed gradle related files

* [chore] add gradle wrapper

* [chore] change gradlew permission to executable

* [chore] update backend dockerfile to expose application port

* [chore] update backend production database configuration

heebong committed:
[feat] add video detail (#27)

* [docs] add index.html, static resourse

* [feat] add videocardtemplate.js

* [chore] junit 5 setting

* [feat] add vo

- Contents, Title, YoutubeId

* [feat] add video entity

* [chore] add database

* [refactor] move column annotation

* [feat]  add const.js

* [feat] add videoCardTemplate method

* [feat] add mainpage event

* [feat] add popular video card section

* [chore] add mockito

* [feat] add POST /videos method

*  최신순으로 video들 가져오는 메서드 구현

* create controllerTest

* [feat] add backend code

* [feat] add video error object

* [refactor] change package structure

* [refactor] remove setter

* [refactor] add RequestMapping

* [refactor] use ResponseEntity

* [refactor] add Contents exception

* [refactor] change exception static value name

* [refactor] add repository annotation

* [refactor] add autowired annotation

* [chore] add trace, debug setting

* [style] formatting

* [feat] add main app

* [feat] add video get api

* [feat] add video not fount exception

* [feat] add video detail page

kangmin46 committed:
[feat] show videos' created date more prettier (#29)

* [feat] impliment date difference in main page, add headerTemplate.js

* [refactor]  uncomment loadDateVideoCards

Laterality committed:
[chore] fix db connection problem of API application (#31)

* Add scripts

- Added initializing script.
- Added docker  image pushing script.
- Added nginx configuration for nginx container.

* Add travis configuration

* refactor: update initializing script remove unnecessary code

* chore: add execution permission to gradlew

* chore: update travis configuration

- Changed branch to trigger build.

* chore: update travis configuration

- Changed branch pattern to trigger build.

* chore: update travis configuration

- Changed build triggering branch name pattern.

* chore: update with travis ci integration

* Added dockerfile for frontend/backend image.
* Added docker-compose file with production/development.
* Added script for each stage(build jar, docker build, docker push).
* Added script for initializing server.
* Added nginx configuration.

* chore: change initialize script's download url temporarily

* chore: update initialzing script typo

* chore: update docker compose file to run nginx container after running app container

* chore: reduce docker compose file

- Updated initializing script.

* chore: update initializing script

* chore: update init script to clone with specific branch

* chore: update dockerfile

* chore: update dockerfile

* chore: update docker compose file typo

* chore: remove unnecessary files

* chore: update docker-compose frontend root directory

* chore: remove unnecessary code

* [chore]: add gitignore

* chore: move gitignore

* chore: update gitignore and missed gradle related files

* [chore] add gradle wrapper

* [chore] change gradlew permission to executable

* [chore] update backend dockerfile to expose application port

* [chore] update backend production database configuration

* [chore] update backend dockerfile to wait until mysql is ready

* [chore] change dockerfile to move entrypoint script to running directory

* [chore] change backend docker entrypoint scripts permission to execute

* [chore] change dockerfile to add --parallel flag to gradle build

* [chore] remove scripts no more needed

wmakerjun commented on frontend/js/common.js:
```diff
@@ -0,0 +1,5 @@
+function getUrlParams() {
+    var params = {};
```
var 대신에 let을 이용해주세요 :)
https://www.daleseo.com/js-var-issues/

heebong replied:
변경했습니다. 확인 부탁드립니다!

wmakerjun commented on frontend/js/mainpage.js:
```diff
@@ -0,0 +1,7 @@
+loadDateVideoCards = function () {
+    api.get("/api/v1/videos?filter=date&page=0&limit=6")
```
와우! fetchapi.js를 만들어서 api 객체를 활용한 점이 대단합니다 💯 
여기서 한발자국만 더 나아가서 api객체를 다룬다면 보다 더 세련되게 코드를 짤수도 있을 것 같아요!!!

```javascript
const Api = function () {
        const request = (url, method, body) => {
            const defaultHeader = {'Content-Type': 'application/json; charset=UTF-8'}
            return fetch(url, {
                method: method,
                headers: defaultHeader,
                body: body
            })
        }

        const createUser = (userBasicInfo) => {
            return request('/api/users', 'post', userBasicInfo)
        }

        return {
            createUser: createUser
        }
    }

  ...

  const api = new Api()
  
  api
    .createUser(JSON.stringify(userBasicInfo))
    .then(response => alert("success"))
    .catch(error => alert(error));

```

api 객체에서 도메인별 api에 대해서 각각 이름을 부여한다면 그 api를 사용하는 function에서의 로직이 더 명확하게 보일 것 같아요~!

heebong replied:
변경했습니다. 확인 부탁드립니다!

wmakerjun commented on frontend/video-create.html:
```diff
@@ -0,0 +1,208 @@
+                                <h4 class="card-title">Video 생성</h4>
+                                <form class="form-horizontal mrg-top-40 pdd-right-30">
+                                    <div class="form-group row">
+                                        <label for="form-1-1" class="col-md-2 control-label">YOUTUBE ID</label>
```
label의 for 속성은 연결되는 form element의 id와 동일해야 합니다. form-1-1 속성은 아래에 있는 input의 'youtube-id'로 통일시켜주세요 :)
https://webdir.tistory.com/319

heebong replied:
변경했습니다. 확인 부탁드립니다!

Laterality committed:
[feat] add login view markup (#32)

* Add scripts

- Added initializing script.
- Added docker  image pushing script.
- Added nginx configuration for nginx container.

* Add travis configuration

* refactor: update initializing script remove unnecessary code

* chore: add execution permission to gradlew

* chore: update travis configuration

- Changed branch to trigger build.

* chore: update travis configuration

- Changed branch pattern to trigger build.

* chore: update travis configuration

- Changed build triggering branch name pattern.

* chore: update with travis ci integration

* Added dockerfile for frontend/backend image.
* Added docker-compose file with production/development.
* Added script for each stage(build jar, docker build, docker push).
* Added script for initializing server.
* Added nginx configuration.

* chore: change initialize script's download url temporarily

* chore: update initialzing script typo

* chore: update docker compose file to run nginx container after running app container

* chore: reduce docker compose file

- Updated initializing script.

* chore: update initializing script

* chore: update init script to clone with specific branch

* chore: update dockerfile

* chore: update dockerfile

* chore: update docker compose file typo

* chore: remove unnecessary files

* chore: update docker-compose frontend root directory

* chore: remove unnecessary code

* chore: move gitignore

* chore: update gitignore and missed gradle related files

* [feat] add login view markup

Laterality committed:
[refactor] reduce unnecessary codes and rename test methods (#34)

* Add scripts

- Added initializing script.
- Added docker  image pushing script.
- Added nginx configuration for nginx container.

* Add travis configuration

* refactor: update initializing script remove unnecessary code

* chore: add execution permission to gradlew

* chore: update travis configuration

- Changed branch to trigger build.

* chore: update travis configuration

- Changed branch pattern to trigger build.

* chore: update travis configuration

- Changed build triggering branch name pattern.

* chore: update with travis ci integration

* Added dockerfile for frontend/backend image.
* Added docker-compose file with production/development.
* Added script for each stage(build jar, docker build, docker push).
* Added script for initializing server.
* Added nginx configuration.

* chore: change initialize script's download url temporarily

* chore: update initialzing script typo

* chore: update docker compose file to run nginx container after running app container

* chore: reduce docker compose file

- Updated initializing script.

* chore: update initializing script

* chore: update init script to clone with specific branch

* chore: update dockerfile

* chore: update dockerfile

* chore: update docker compose file typo

* chore: remove unnecessary files

* chore: update docker-compose frontend root directory

* chore: remove unnecessary code

* chore: move gitignore

* chore: update gitignore and missed gradle related files

* [feat] refactor exception handling and returning ResponseEntity

* [feat] refactor backend tests

- Formatted code.
- Removed empty class.
- Renamed `Error` to `ErrorResponse` to distinguish with `java.lang.Error `.
- Renamed test method names consistently.
- Changed video list retrieval with views filter to throw error as currently not implemented.

wmakerjun commented on frontend/js/video/videocardtemplate.js:
```diff
@@ -0,0 +1,50 @@
+    }
+}
+
+function calculateDate(responseDate) {
```
우와 시간표시를 하기 위해 꼼꼼하게 구현한 점 너무 멋집니다 💯 
혹시 시간과 관련된 표시를 위해서 다른 기능을 추가하게 된다면 [momentjs](https://momentjs.com) 라이브러리를 활용하면 더욱 쉽고 빠르게 구현할 수도 있어요~! 참고해주길 바래요 :)

heebong replied:
변경했습니다. 확인 부탁드립니다!

wmakerjun commented on frontend/js/video/videocardtemplate.js:
```diff
@@ -0,0 +1,50 @@
+function addVideoCardTemplate(data, number, categorie) {
+    const dateDifference = calculateDate(data[number].createDate)
+    const vidioCardTemplate = `<div class="col-lg-2 padding-2">
```
**Vidio**CardTemplate에서 Vidio는 오타겠죠? ㅎㅎ
html template 부분에서도 인덴트를 줘서 관리하는게 구조를 파악하고 수정하는데 더욱 용이하답니다. :)
template 파일로 따로 분리해도 좋구요~!

heebong replied:
변경했습니다. 확인 부탁드립니다!

Laterality committed:
[refactor] solve issue #35 (#37)

* Add scripts

- Added initializing script.
- Added docker  image pushing script.
- Added nginx configuration for nginx container.

* Add travis configuration

* refactor: update initializing script remove unnecessary code

* chore: add execution permission to gradlew

* chore: update travis configuration

- Changed branch to trigger build.

* chore: update travis configuration

- Changed branch pattern to trigger build.

* chore: update travis configuration

- Changed build triggering branch name pattern.

* chore: update with travis ci integration

* Added dockerfile for frontend/backend image.
* Added docker-compose file with production/development.
* Added script for each stage(build jar, docker build, docker push).
* Added script for initializing server.
* Added nginx configuration.

* chore: change initialize script's download url temporarily

* chore: update initialzing script typo

* chore: update docker compose file to run nginx container after running app container

* chore: reduce docker compose file

- Updated initializing script.

* chore: update initializing script

* chore: update init script to clone with specific branch

* chore: update dockerfile

* chore: update dockerfile

* chore: update docker compose file typo

* chore: remove unnecessary files

* chore: update docker-compose frontend root directory

* chore: remove unnecessary code

* chore: move gitignore

* chore: update gitignore and missed gradle related files

* [refactor] move codes commonly used into one context object

- Replaced double quotes(") in js to single quotes(').

Laterality committed:
[chore] solve #36 (#40)

* Add scripts

- Added initializing script.
- Added docker  image pushing script.
- Added nginx configuration for nginx container.

* Add travis configuration

* refactor: update initializing script remove unnecessary code

* chore: add execution permission to gradlew

* chore: update travis configuration

- Changed branch to trigger build.

* chore: update travis configuration

- Changed branch pattern to trigger build.

* chore: update travis configuration

- Changed build triggering branch name pattern.

* chore: update with travis ci integration

* Added dockerfile for frontend/backend image.
* Added docker-compose file with production/development.
* Added script for each stage(build jar, docker build, docker push).
* Added script for initializing server.
* Added nginx configuration.

* chore: change initialize script's download url temporarily

* chore: update initialzing script typo

* chore: update docker compose file to run nginx container after running app container

* chore: reduce docker compose file

- Updated initializing script.

* chore: update initializing script

* chore: update init script to clone with specific branch

* chore: update dockerfile

* chore: update dockerfile

* chore: update docker compose file typo

* chore: remove unnecessary files

* chore: update docker-compose frontend root directory

* chore: remove unnecessary code

* chore: move gitignore

* chore: update gitignore and missed gradle related files

* [chore] segregate application properties for productions from development environment

* [chore] added application output directory(db_data, nginx) into gitignore

* [docs] update README to guide run project

Laterality committed:
[docs] update readme (#41)

* Add scripts

- Added initializing script.
- Added docker  image pushing script.
- Added nginx configuration for nginx container.

* Add travis configuration

* refactor: update initializing script remove unnecessary code

* chore: add execution permission to gradlew

* chore: update travis configuration

- Changed branch to trigger build.

* chore: update travis configuration

- Changed branch pattern to trigger build.

* chore: update travis configuration

- Changed build triggering branch name pattern.

* chore: update with travis ci integration

* Added dockerfile for frontend/backend image.
* Added docker-compose file with production/development.
* Added script for each stage(build jar, docker build, docker push).
* Added script for initializing server.
* Added nginx configuration.

* chore: change initialize script's download url temporarily

* chore: update initialzing script typo

* chore: update docker compose file to run nginx container after running app container

* chore: reduce docker compose file

- Updated initializing script.

* chore: update initializing script

* chore: update init script to clone with specific branch

* chore: update dockerfile

* chore: update dockerfile

* chore: update docker compose file typo

* chore: remove unnecessary files

* chore: update docker-compose frontend root directory

* chore: remove unnecessary code

* chore: move gitignore

* chore: update gitignore and missed gradle related files

* [docs] update readme

wmakerjun reviewed (changes requested):
기능에 따라 자바스크립트 파일들을 분리한게 매우 인상적입니다. 💯 
코멘트 남겨준 것들 약간만 수정해준다면 코드가 더 세련되게 될 것 같아요 너무 수고하셨습니다 😃

Laterality committed:
fix #42 (#43)

* Add scripts

- Added initializing script.
- Added docker  image pushing script.
- Added nginx configuration for nginx container.

* Add travis configuration

* refactor: update initializing script remove unnecessary code

* chore: add execution permission to gradlew

* chore: update travis configuration

- Changed branch to trigger build.

* chore: update travis configuration

- Changed branch pattern to trigger build.

* chore: update travis configuration

- Changed build triggering branch name pattern.

* chore: update with travis ci integration

* Added dockerfile for frontend/backend image.
* Added docker-compose file with production/development.
* Added script for each stage(build jar, docker build, docker push).
* Added script for initializing server.
* Added nginx configuration.

* chore: change initialize script's download url temporarily

* chore: update initialzing script typo

* chore: update docker compose file to run nginx container after running app container

* chore: reduce docker compose file

- Updated initializing script.

* chore: update initializing script

* chore: update init script to clone with specific branch

* chore: update dockerfile

* chore: update dockerfile

* chore: update docker compose file typo

* chore: remove unnecessary files

* chore: update docker-compose frontend root directory

* chore: remove unnecessary code

* chore: move gitignore

* chore: update gitignore and missed gradle related files

* [docs] update readme

* [fix] solve issue

Laterality committed:
[docs] update readme with guide for windows environment (#44)

* [docs] update readme to guide when get error in windows environment

Laterality committed:
Update README.md

kangmin46 committed:
[fontend feedback]  (#48)

* [feat] impliment date difference in main page, add headerTemplate.js

* [refactor]  uncomment loadDateVideoCards

* [refactor]  change variable name, change method name

kimjungmin-developer committed:
[feat] add frontend facilities for video edit (#50)

* [feat] add frontend facilities for video edit

* [fix] fix json error

* [fix] Fix html error

heebong committed:
[feedback] frontend refactor (#45) (#46)

* [refactor] remove form tag

- remove form tag
- change label tag to div tag

* [refactor] add form tag

Laterality committed:
[fix] index.html header event listener issue(#51) (#52)

* Add scripts

- Added initializing script.
- Added docker  image pushing script.
- Added nginx configuration for nginx container.

* Add travis configuration

* refactor: update initializing script remove unnecessary code

* chore: add execution permission to gradlew

* chore: update travis configuration

- Changed branch to trigger build.

* chore: update travis configuration

- Changed branch pattern to trigger build.

* chore: update travis configuration

- Changed build triggering branch name pattern.

* chore: update with travis ci integration

* Added dockerfile for frontend/backend image.
* Added docker-compose file with production/development.
* Added script for each stage(build jar, docker build, docker push).
* Added script for initializing server.
* Added nginx configuration.

* chore: change initialize script's download url temporarily

* chore: update initialzing script typo

* chore: update docker compose file to run nginx container after running app container

* chore: reduce docker compose file

- Updated initializing script.

* chore: update initializing script

* chore: update init script to clone with specific branch

* chore: update dockerfile

* chore: update dockerfile

* chore: update docker compose file typo

* chore: remove unnecessary files

* chore: update docker-compose frontend root directory

* chore: remove unnecessary code

* chore: move gitignore

* chore: update gitignore and missed gradle related files

* [fix] changed script tag order

* [refactor] remove unnecessary logging code

heebong committed:
[fix] change base url (#55)

heebong committed:
[docs] update README

- change BASE_URL location

heebong committed:
[refactor] change video detail markup (#56)

Laterality committed:
[fix] fix issue through add 'type' attribute to each button element (#58)

kang-hyungu commented on backend/Dockerfile:
```diff
@@ -0,0 +1,18 @@
+RUN ./gradlew --parallel build
+RUN mv ./build/libs/$JAR_NAME /usr/src/app
+RUN mv ./docker-entrypoint.sh /usr/src/app
+RUN mv ./src/main/resources/application-production.yml /usr/src/app/application.yml
```
application.yml 파일명을 변경하는 방법도 있지만 아래와 같이 스프링 띄울 때 어느 설정 파일을 사용할지 정하는 방법도 있어요
$ java -jar -Dspring.profiles.active=production demo-0.0.1-SNAPSHOT.jar
spring boot active profile이라는 검색어로 구글링해보시면 도움이 될거에요

kang-hyungu commented on backend/build.gradle:
```diff
@@ -0,0 +1,59 @@
+    }
+    testImplementation 'org.junit.jupiter:junit-jupiter-params:5.4.2'
+    testImplementation 'io.projectreactor:reactor-test'
+    testImplementation 'org.springframework.restdocs:spring-restdocs-mockmvc'
```
spring rest docs는 안쓰는 것 같은데... 불필요한 의존 라이브러리는 제거해보아요

kang-hyungu commented on backend/build.gradle:
```diff
@@ -0,0 +1,59 @@
+plugins {
+    id 'org.asciidoctor.convert' version '1.5.3'
```
asciidoctor도 쓰는 곳이 보이진 않네요

kang-hyungu commented on backend/src/main/java/com/woowacourse/edd/application/converter/VideoConverter.java:
```diff
@@ -0,0 +1,32 @@
+
+import java.time.format.DateTimeFormatter;
+
+@Component
```
스프링 빈 컨테이너에서 관리할 필요가 있을까요?

kang-hyungu commented on backend/src/main/java/com/woowacourse/edd/application/service/VideoService.java:
```diff
@@ -0,0 +1,56 @@
+        this.videoConverter = videoConverter;
+    }
+
+    public VideoResponse save(VideoSaveRequestDto requestDto) {
```
transaction 어노테이션이 필요해보이네요

kang-hyungu commented on backend/src/main/java/com/woowacourse/edd/application/service/VideoService.java:
```diff
@@ -0,0 +1,56 @@
+        return videoConverter.toResponse(video);
+    }
+
+    public List<VideoPreviewResponse> findVideosByDate(int page, int limit) {
```
1. 메소드명은 ByDate인데 검색 조건은 페이징이네요. 정렬조건이 CreateDate이어서 Date을 붙인 것 같네요
2. PageRequest를 컨트롤러에서 전달 받는게 낫지 않을까요? 다른 정렬조건을 붙이고 싶으면 서비스에 메서드를 새로 추가해야되는 구조네요
3. 응답은 Page<VideoPreviewResponse>로 전달하면 API를 사용하는 클라이언트 입장에서 더 많은 정보를 얻을 수 있지 않을까요? 첫 번째 페이지나 마지막 페이지라는 정보 등등..

Conatuseus replied:
프론트에서 간단히 요청해서 가져오도록 하고 싶어서 그렇게 했었어요.
저희가 구현한 방법대로 하면 정렬조건에 따라 많은 메서드가 생기는 걸 느꼈습니다.
해당 메서드는 Pageable을 컨트롤러에 전달받도록 했습니다.

kang-hyungu commented on backend/src/main/java/com/woowacourse/edd/application/service/VideoService.java:
```diff
@@ -0,0 +1,56 @@
+        PageRequest pageRequest = PageRequest.of(page, limit, Sort.by(CREATE_DATE).descending());
+        Page<Video> foundVideos = videoRepository.findAll(pageRequest);
+
+        return foundVideos.getContent().stream()
```
```suggestion
        return foundVideos.map(videoConverter::toPreviewResponse)
                .getContent();
```

응답을 Page<VideoPreviewResponse>로 변경하면 getContent도 필요없을거에요

Conatuseus replied:
Page로 변경해서 getcontent 삭제했습니다.

kang-hyungu commented on backend/src/main/java/com/woowacourse/edd/domain/Video.java:
```diff
@@ -0,0 +1,69 @@
+    private Long id;
+
+    @Embedded
+    private YoutubeId youtubeId;
```
YoutubeId, Title, Contents를 vo로 따로 분리할 필요가 있을까요?
vo로 분리했을 때 장점이 현재는 안보이네요

kang-hyungu commented on backend/src/main/java/com/woowacourse/edd/presentation/controller/VideoController.java:
```diff
@@ -0,0 +1,46 @@
+
+    @GetMapping
+    public ResponseEntity<List<VideoPreviewResponse>> findVideosByFilter(@RequestParam String filter, @RequestParam int page, @RequestParam int limit) {
+        if (DATE.equals(filter)) {
```
1. 필터면 검색조건일텐데 Date은 정렬조건 아닌가요? 메소드명의 의미가 잘못 전달되겠네요
2. 검색 조건을 추가를 위해 컨트롤러에서 if절로 필터하다보면 검색 조건이 늘어나면서 if 절이 많아지고 서비스 메서드도 계속 추가되겠네요. 검색 조건 정보를 담은 객체를 서비스로 넘겨서 처리하는건 어떨까요? jpa specification라는 기술이 있는데 지금 적용하기엔 과하니 서비스에서 적절히 분기처리하거나 querydsl에서 분기처리하기도 해요. 나름의 방법을 찾아서 코드 복잡도를 제거해보세요

Conatuseus replied:
<< 지금은 Pageable로 변경해서 위 내용이 사라졌습니다. >>

if절을 삭제하기 위해 전략패턴이나 그런 것들을 사용하려고 했지만 service layer에 interface를 생성해서 사용해도 되나 라는 의문이 생겨서 해결하지 못했습니다.
jpa specification은 다음에 기회가 되면 공부해서 이런 상황에 사용해보겠습니다!

kang-hyungu commented on backend/src/main/java/com/woowacourse/edd/presentation/exceptionhandler/VideoExceptionHandler.java:
```diff
@@ -0,0 +1,18 @@
+    @ResponseBody
+    @ExceptionHandler(ErrorResponseException.class)
+    public ResponseEntity<ErrorResponse> handleErrorResponse(ErrorResponseException e) {
+        return ResponseEntity.badRequest()
```
ErrorResponseException을 상속하는 모든 예외는 http response status code를 bad request로 응답하는게 맞을까요?
상황에 따라 400, 401, 404 등으로 나뉘지 않을까요?

kang-hyungu commented on backend/src/main/resources/application-production.yml:
```diff
@@ -0,0 +1,19 @@
+      enabled: false
+  jpa:
+    hibernate:
+      ddl-auto: update
```
운영은 무조건 none으로 처리하세요
변경된 엔티티를 자동으로 적용하는 설정은 무척 위험합니다. update도 위험합니다. 의도치 않게 운영에 반영됐을 때 변경된 db 스키마를 되돌리기 어려울 수 있습니다.

kang-hyungu commented on backend/src/test/java/com/woowacourse/edd/EddApplicationTests.java:
```diff
@@ -0,0 +1,19 @@
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@AutoConfigureWebTestClient
+@ExtendWith(SpringExtension.class)
```
AutoConfigureWebTestClient, ExtendWith 둘다 필요없는 어노테이션으로 보이네요

heebong replied:
ControllerTests가 EddApplicationTests를 상속받아서 EddApplicationTests에 해당 어노테이션을 추가했습니다. 지금은 ControllerTests가 하나지만 앞으로 더 늘어날 때를 대비해서 상속받는 형식으로 사용했는데 바꾸는게 좋을까요??

kang-hyungu replied:
SpringBootTest 어노테이션에 포함된 어노테이션들이라 불필요하다고 생각했어요
AutoConfigureWebTestClient는 따로 추가해야될 수도 있어요

kang-hyungu commented on backend/src/test/java/com/woowacourse/edd/application/service/VideoServiceTest.java:
```diff
@@ -0,0 +1,51 @@
+        Video video = new Video(new YoutubeId("1234"), new Title("title"), new Contents("contents"));
+        VideoResponse videoResponse = new VideoResponse(100L, "1234", "title", "contents", getFormedDate());
+        when(videoRepository.save(any())).thenReturn(video);
+        when(videoConverter.toEntity(any())).thenReturn(video);
```
컨버터는 굳이 mocking할 필요 없지 않나요?

kang-hyungu commented on backend/src/test/java/com/woowacourse/edd/application/service/VideoServiceTest.java:
```diff
@@ -0,0 +1,51 @@
+        when(videoConverter.toEntity(any())).thenReturn(video);
+        when(videoConverter.toResponse(any())).thenReturn(videoResponse);
+
+        assertThat(service.save(new VideoSaveRequestDto("1234", "title", "contents"))).isEqualTo(videoResponse);
```
<img width="1079" alt="image" src="https://user-images.githubusercontent.com/15669435/63211203-10bdab00-c12f-11e9-9d29-913c9482d887.png">

kang-hyungu replied:
스샷 보시면 save 변수와 videoResponse 변수가 가리키는 객체가 동일합니다.
동일한(identity) 객체가 아니라 동등한(equality) 객체인지 테스트해봐야하지 않을까요?

kang-hyungu commented on backend/src/test/java/com/woowacourse/edd/domain/VideoTest.java:
```diff
@@ -0,0 +1,15 @@
+public class VideoTest {
+    @Test
+    void create_video() {
+        assertDoesNotThrow(() -> new Video(new YoutubeId("sadfsadfasdf"), new Title("title"), new Contents("contents")));
```
vo가 많아지면 테스트 코드 작성도 불편해져요
vo가 특별히 하는 일이 없다면 객체로 잘게 쪼개는게 실무에선 독이 될수도 있어요

kang-hyungu commented on backend/src/test/java/com/woowacourse/edd/presentation/controller/VideoControllerTests.java:
```diff
@@ -0,0 +1,140 @@
+    void save() {
+        VideoSaveRequestDto videoSaveRequestDto = new VideoSaveRequestDto(DEFAULT_VIDEO_YOUTUBEID, DEFAULT_VIDEO_TITLE, DEFAULT_VIDEO_CONTENTS);
+
+        webTestClient.post().uri("/v1/videos")
```
webTestClient를 호출 할 때 중복되는 부분을 제거해보면 어떨까요

kang-hyungu commented on backend/src/test/java/com/woowacourse/edd/presentation/controller/VideoControllerTests.java:
```diff
@@ -0,0 +1,140 @@
+            .jsonPath("$.message").isEqualTo("그런 비디오는 존재하지 않아!");
+    }
+
+    private String getFormedDate() {
```
테스트 코드에 중복되서 나오네요.
중복을 제거하려면 다른 클래스를 추가할 수도 있겠네요

kang-hyungu reviewed (changes requested):
안녕하세요 edd팀 여러분!
도커를 도입하고 프론트, 백엔드로 소스코드를 구분해서 개발 진행하는게 인상 깊네요 👍 
백엔드에 몇 가지 피드백 남겼으니 참고하세요!
궁금한 점 있으면 언제든 dm 주세요

gracefulBrown commented on backend/src/main/java/com/woowacourse/edd/application/converter/VideoConverter.java:
```diff
@@ -0,0 +1,32 @@
+        return new VideoResponse(video.getId(), video.getYoutubeId().getYoutubeId(), video.getTitle().getTitle(), video.getContents().getContents(), date);
+    }
+
+    public VideoPreviewResponse toPreviewResponse(Video video) {
```
중복제거를 시도해보면 좋을 것 같네요 :)

gracefulBrown commented on backend/src/main/java/com/woowacourse/edd/application/response/VideoResponse.java:
```diff
@@ -0,0 +1,38 @@
+
+public class VideoResponse {
+
+    private final long id;
```
클래스 변수에 final을 선언한 클래스가 있고 선언하지 않은 클래스가 있는데 의도된 내용인가요?
만약 아니라면 final 사용 관련 컨벤션을 맞춰주는것도 좋을 것 같습니다 :)

gracefulBrown commented on backend/src/main/java/com/woowacourse/edd/domain/Video.java:
```diff
@@ -0,0 +1,69 @@
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
```
VideoResponse에는 id의 타입이 long을 사용했는데요 의도된 내용인가요?

gracefulBrown commented on backend/src/main/java/com/woowacourse/edd/presentation/controller/VideoController.java:
```diff
@@ -0,0 +1,46 @@
+    @PostMapping
+    public ResponseEntity<VideoResponse> saveVideo(@RequestBody VideoSaveRequestDto requestDto) {
+        VideoResponse response = videoService.save(requestDto);
+        return ResponseEntity.ok(response);
```
적절한 http status code 를 고민해보면 좋을 것 같아요 :)

gracefulBrown commented on backend/build.gradle:
```diff
@@ -0,0 +1,59 @@
+    set('snippetsDir', file("build/generated-snippets"))
+}
+
+dependencies {
```
현재 빌드 시 실패하는 테스트가 있는경우에도 배포가 되고 있는데요
테스트가 실패할 경우 빌드는 실패가 되어야 합니다.

빌드나 배포할 때 테스트가 수행되는 지 확인해보셨나요?
혹시 테스트가 수행되지 않는다면 아래의 링크를 참고해서 테스트를 수행해보세요!
https://www.baeldung.com/junit-5-gradle

gracefulBrown reviewed (approved):
안녕하세요! 브라운입니다.
프로젝트를 잘 수행해주시고 있군요 👍 
몇가지 참고하면 좋은 부분 코멘트로 남겼으니 확인해보시고 진행해주세요 :)
(참고로 실패하는 테스트가 있는경우 빌드가 되면 안되는 점 반드시 인지해주시면 좋을 것 같아요~)

Laterality committed:
[docs] update README.md

- Corrected wrong command for windows user

Laterality committed:
백엔드 피드백 반영(#59) (#63)

* [feat] add db schema as sql file for initialize mysql container

* [chore] change API application to use active profile instaed of renaming properties file

* [chore] changed docker-compose file to initialize database scheme

* [refactor] update build.gradle

- Removed dependencies not used currently.
- Configured to test when build jar file.

* [chore] update API application properties for production

- Changed ddl generation to 'none'.
- Changed jdbc driver class for MySql.
- Added logging property not to show condition evaluation report.

* [refactor] remove unnecessary annotation

* [refactor] add transactional annotation to service

* [refactor] change exceptions to have own status code

* [refactor] refactor video service test

* [refactor] update test as exception handling is changed

heebong committed:
[test] duplicate controller test (#62)

* [test] duplicate controller test

* [test] change tab space

- 8 space -> 4 space

* [test] rename function

* [test] add util class

* [test] add utils class

* [test] change form

* [test] rename fail function

* [test] change test function

Conatuseus committed:
백엔드 피드백 반영(#64) (#66)

* [refactor] fix method for find videos by page

* [refactor] delete duplicated, change VideoPreviewResonse location

* [refactor] delete duplicated, ResponseEntity to builder

* [refactor] code format

* [refactor] fix return type of saveVideo method

Conatuseus committed:
[Page 기능] 수정해서 PR보냅니다. (#69)

* [refactor] fix method for find videos by page

* [refactor] delete duplicated, change VideoPreviewResonse location

* [refactor] delete duplicated, ResponseEntity to builder

* [refactor] code format

* [refactor] fix return type of saveVideo method

* [refactor] fix parameter, testCode

* [refactor] Reformat code

* [refactor] fix type of id to Long

* [refactor] change method for getting videos

* [refactor] change test size

* Delete $PRODUCT_WORKSPACE_FILE$

* Delete .gitignore

* Delete checkstyle-idea.xml

* Delete misc.xml

* Delete modules.xml

* Delete vcs.xml

heebong committed:
[refactor] add feedback (#68)

* [refactor] change save http status 200 -> 201

* [refactor] add final

* [test] add acceptance test

* [refactor] change test code

Laterality committed:
[fix] fix wrong href of header profile image (#71)

Laterality committed:
[fix] fix issue (#73)

kang-hyungu reviewed (approved):
피드백 반영하느라 고생하셨어요!
머지 처리할게요!
